### ISSUE TITLE
[BUG] TreeSitter compatability for LaTeX environments broken

### ISSUE THREAD

**Issue by average-sphere-spectrum-enjoyer** [ellisonleao/gruvbox.nvim#329]
### Describe the bug

Commit 99ddf57 of nvim-treesitter breaks the coloring of the "environments" in LaTeX. To reproduce the bug:
1. Use a commit of nvim-treesitter newer than 99ddf57.
2. Open a .tex file with \begin{_environment_} ... \end{_environment_}

An example could be the file bug.tex:

```latex
\begin{document}

hello world

\end{document}
```

It currently looks like (with transparent background):
![incorrect](https://github.com/ellisonleao/gruvbox.nvim/assets/166172826/39b292ce-54a8-4e49-a357-1de988ad7e63)

### Expected behaviour

It should look something like this (this is what it looks like when using the parent commit of 99ddf57 - again, with transparent background):

![correct](https://github.com/ellisonleao/gruvbox.nvim/assets/166172826/70a114e6-23f9-4d12-8a86-8b92bbd67aa9)

This has quite a big influence on the UX as one uses environments in LaTeX constantly. For a "bigger" example see the correct:
![correct-full](https://github.com/ellisonleao/gruvbox.nvim/assets/166172826/d3ab3969-0750-405f-b0a1-646839c19292)
and incorrect:
![incorrect-full](https://github.com/ellisonleao/gruvbox.nvim/assets/166172826/ecdf12ba-72c1-409c-9f88-4075ddd76fc1)

### Screenshots

It should work as..

**Comment by average-sphere-spectrum-enjoyer** [comment]
Reading the commit message we get "Language-specific subcaptures should instead be added in user config or
a custom language plugin".
I don't know how treesitter and colorschemes work, but I guess it should be possible to simply copy the earlier @markup.environment to some builtin version?

**Comment by ellisonleao** [comment]
@average-sphere-spectrum-enjoyer yes, you probably need to override either the lsp semantic highlights or the treesitter ones. I am closing this one but if you have a fix for this, please open a PR so we can discuss it over there.
